trieste/models/gpflow/sampler.py

```diff
@@ -170,13 +212,26 @@ def sample(self, at: TensorType, *, jitter: float = DEFAULTS.JITTER) -> TensorTy
 
         mean, cov = self._model.predict_joint(at)  # [..., B, L], [..., L, B, B]
 
-        if not self._initialized:
-            self._eps.assign(
-                tf.random.normal(
+        def sample_eps() -> tf.Tensor:
+            self._initialized.assign(True)
+            if self._qmc:
+                skip = IndependentReparametrizationSampler.skip
+                IndependentReparametrizationSampler.skip.assign(skip + self._sample_size)
+                normal_samples = qmc_normal_samples(
```

Comment:
I think this is where the issue is. Fundamentally you want the Sobol sequence to be of dimension `B = batch_size`. That is, each sample will be a multi-output Gaussian of dimension `batch_size` and you want to be space-filling in this space. 

Then you want `S = self._sample_size` samples = number of iterates from the Sobol sequence.

Finally there is the number of latents, `L = mean.shape[-1]`. 

Currently this code calls: `tf.math.sobol_sample(dim=[L], num_results=B*S)`.

I think what you want is `tf.math.sobol_sample(dim=[B, L], num_results=S)`.

Reply:
@uri-granta I agree with @vpicheny above on `tf.math.sobol_sample(dim=[B], num_results=S*L)`, but what is implemented below is not correct, we have at the moment `tf.math.sobol_sample(dim=[B*L], num_results=S)`, right?

Reply:
You're totally right, oops! Here's a fix PR: https://github.com/secondmind-labs/trieste/pull/713